models/cnn/AlexNet/test_AlexNet.py

```diff
@@ -29,7 +30,148 @@ def test_overfit(model):
+    test_overfit(model)
+    #num_Classes = 9
+    #case 1 
+    model = AlexNet(num_classes=9)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=9)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=9)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 8
+    #case 1 
+    model = AlexNet(num_classes=8)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=8)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=8)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 7
+    #case 1 
+    model = AlexNet(num_classes=7)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=7)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=7)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 6
+    #case 1 
+    model = AlexNet(num_classes=6)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=6)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=6)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 5
+    #case 1 
+    model = AlexNet(num_classes=5)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=5)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=5)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 4
+    #case 1 
+    model = AlexNet(num_classes=4)
+    model = model.build(input_shape=(128, 128, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 2 
+    model = AlexNet(num_classes=4)
+    model = model.build(input_shape=(225, 225, 3))
+    test_forward_pass(model)
+    test_overfit(model)
+    #case 3
+    model = AlexNet(num_classes=4)
+    model = model.build(input_shape=(128, 128, 1))
+    test_forward_pass(model)
+    test_overfit(model)
+    #num_Classes = 3
```

Comment:
No need to run too much test cases. These test cases are all equivalent. If one fails then all fails. If one passes then you know that all other can pass.